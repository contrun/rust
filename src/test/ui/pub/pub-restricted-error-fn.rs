pub(crate) () fn foo() {}
//~^ ERROR visibility `pub(crate)` is not followed by an item
//~| ERROR statements cannot reside in modules
//~| ERROR `main` function not found
