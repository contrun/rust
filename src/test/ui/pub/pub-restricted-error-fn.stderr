error: visibility `pub(crate)` is not followed by an item
  --> $DIR/pub-restricted-error-fn.rs:1:1
   |
LL | pub(crate) () fn foo() {}
   | ^^^^^^^^^^ the visibility
   |
   = help: you likely meant to define an item, e.g., `pub(crate) fn foo() {}`

error: statements cannot reside in modules
  --> $DIR/pub-restricted-error-fn.rs:1:12
   |
LL | pub(crate) () fn foo() {}
   |            ^^
   |
   = note: the program entry point starts in `fn main() { ... }`, defined in `main.rs`
   = note: for more on functions and how to structure your program, see https://doc.rust-lang.org/book/ch03-03-how-functions-work.html
help: consider moving the statements into a function
   |
LL | pub(crate) fn my_function() -> _ { () } fn foo() {}
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0601]: `main` function not found in crate `pub_restricted_error_fn`
  --> $DIR/pub-restricted-error-fn.rs:1:1
   |
LL | pub(crate) () fn foo() {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^ consider adding a `main` function to `$DIR/pub-restricted-error-fn.rs`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0601`.
